scripts/trading_db.py

```
@@ -0,0 +1,126 @@
+    with conn_manager.context() as conn:
+        cur = conn.execute("SELECT id, name, definition FROM rules")
+        return [dict(row) for row in cur.fetchall()]
+
+
+def add_strategy(name: str, rule_ids: Sequence[int], conn_manager: ConnectionManager) -> int:
+    """Insert a strategy composed of rule IDs."""
+    # Verify all rule IDs exist
+    with conn_manager.context() as conn:
+        qs = ",".join("?" * len(rule_ids))
+        cur = conn.execute(
+            f"SELECT id FROM rules WHERE id IN ({qs})",
```

Ruff (0.11.9)

Possible SQL injection vector through string-based query construction

(S608)